exporters/otlpjson/src/test/java/io/opentelemetry/exporter/otlpjson/OtlpJsonLoggingMetricExporterTest.java

```diff
@@ -0,0 +1,163 @@
+          public void close() {}
+        };
+    logger.addHandler(handler);
+    logger.setUseParentHandlers(false);
+    try {
+      exporter.export(Arrays.asList(METRIC1, METRIC2));
+
+      assertThat(logged)
+          .hasSize(1)
+          .allSatisfy(log -> assertThat(log.getLevel()).isEqualTo(Level.INFO));
+      JSONAssert.assertEquals(
+          "{\n"
+              + "  \"resource\": {\n"
+              + "    \"attributes\": [{\n"
+              + "      \"key\": \"key\",\n"
+              + "      \"value\": {\n"
+              + "        \"stringValue\": \"value\"\n"
+              + "      }\n"
+              + "    }]\n"
+              + "  },\n"
+              + "  \"instrumentationLibraryMetrics\": [{\n"
+              + "    \"instrumentationLibrary\": {\n"
+              + "      \"name\": \"instrumentation2\",\n"
+              + "      \"version\": \"2\"\n"
+              + "    },\n"
+              + "    \"metrics\": [{\n"
+              + "      \"name\": \"metric2\",\n"
+              + "      \"description\": \"metric2 description\",\n"
+              + "      \"unit\": \"s\",\n"
+              + "      \"doubleSum\": {\n"
+              + "        \"dataPoints\": [{\n"
+              + "          \"labels\": [{\n"
+              + "            \"key\": \"cat\",\n"
+              + "            \"value\": \"meow\"\n"
+              + "          }],\n"
+              + "          \"startTimeUnixNano\": \"1\",\n"
+              + "          \"timeUnixNano\": \"2\",\n"
+              + "          \"value\": 4.0\n"
+              + "        }],\n"
+              + "        \"aggregationTemporality\": \"AGGREGATION_TEMPORALITY_CUMULATIVE\",\n"
+              + "        \"isMonotonic\": true\n"
+              + "      }\n"
+              + "    }]\n"
+              + "  }, {\n"
+              + "    \"instrumentationLibrary\": {\n"
+              + "      \"name\": \"instrumentation\",\n"
+              + "      \"version\": \"1\"\n"
+              + "    },\n"
+              + "    \"metrics\": [{\n"
+              + "      \"name\": \"metric1\",\n"
+              + "      \"description\": \"metric1 description\",\n"
+              + "      \"unit\": \"m\",\n"
+              + "      \"doubleSum\": {\n"
+              + "        \"dataPoints\": [{\n"
+              + "          \"labels\": [{\n"
+              + "            \"key\": \"cat\",\n"
+              + "            \"value\": \"meow\"\n"
+              + "          }],\n"
+              + "          \"startTimeUnixNano\": \"1\",\n"
+              + "          \"timeUnixNano\": \"2\",\n"
+              + "          \"value\": 4.0\n"
+              + "        }],\n"
+              + "        \"aggregationTemporality\": \"AGGREGATION_TEMPORALITY_CUMULATIVE\",\n"
+              + "        \"isMonotonic\": true\n"
+              + "      }\n"
+              + "    }]\n"
+              + "  }]\n"
+              + "}",
+          logged.get(0).getMessage(),
```

Comment:
If I read this right...the log message will be split across multiple lines? Don't we want to make sure that every log message is only a single line, so it can work with normal logging systems?

Reply:
Thanks for noticing! This is a JSON assert so whitespace is ignored. I added additional checks that the message has no newlines and realized I needed to set that.